index.js

```diff
@@ -37,35 +80,110 @@ const show = ({ user, repo, branch }) => `https://github.com/${user}/${repo}/tre
+      console.error(chalk`Case 2. You specified {yellow wrong branches} and/or repos.`);
+      console.error(chalk`    {dim Solution: Double check the command line and fix the error.}`);
+      console.error(chalk`Case 3. A branch has {yellow already been deleted}.`);
+      console.error(chalk`    {dim Solution: No need to worry.}`);
+    } else if (e.response.data.message === 'Reference does not exist') {
+      console.error(chalk`{red Failed: {italic ${e.response.data.message}.}} {dim Possible reasons:}`);
+      console.error(chalk`Case 1. You specified {yellow wrong branches} and/or repos.`);
+      console.error(chalk`    {dim Solution: Double check the command line and fix the error.}`);
+      console.error(chalk`Case 2. You want to re-create a branch that {yellow has been deleted} before.`);
+      console.error(chalk`    {dim Solution: Use {reset --create}.}`);
+      console.error(chalk`Case 3. The upstream repo developers {yellow created a new branch}.`);
+      console.error(chalk`    {dim Solution: Use {reset --create}.}`);
+    } else if (e.response.data.message === 'Reference already exists') {
+      console.error(chalk`{red Failed: {italic ${e.response.data.message}.}} {dim Possible reasons:}`);
+      console.error(chalk`Case 1. You specified {yellow wrong branches} and/or repos.`);
+      console.error(chalk`    {dim Solution: Double check the command line and fix the error.}`);
+      console.error(chalk`Case 2. You want to create a branch that has {yellow already been created}.`);
+      console.error(chalk`    {dim Solution: No need to worry.}`);
+      console.error(chalk`Case 3. You want to {yellow update} an exiting branch, not creating a new one.`);
+      console.error(chalk`    {dim Solution: Do {reset NOT} use {reset --create}.}`);
+    } else if (e.response.data.message === 'Update is not a fast forward') {
+      console.error(chalk`{red Failed: {italic ${e.response.data.message}.}} {dim Possible reasons:}`);
+      console.error(chalk`Case 1. You specified {yellow wrong branches} and/or repos.`);
+      console.error(chalk`    {dim Solution: Double check the command line and fix the error.}`);
+      console.error(chalk`Case 2. {yellow Your branch is ahead} of upstream (i.e. you have some changes that upstream doesn't).`);
+      console.error(chalk`    {dim Solution: hub-sync won't help in this case. Refer to this guide:}`);
+      console.error(chalk`    {dim {underline https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork}}`);
+      console.error(chalk`Case 3. The upstream repo developers {yellow force-pushed {italic their} branches}.`);
+      console.error(chalk`    {dim Solution: If you are 100% sure what you are doing, go for {reset --force}. Be extra extra careful in this case.}`);
+      console.error(chalk`    {dim You {reset will} lose data if the case is not handled carefully.}`);
+    } else {
```

Comment:
If this is the exception handling, these could be the best implementations I've seen so far.